src/pages/checkout/payment-method.js

```diff
@@ -0,0 +1,234 @@
+    this.payPalCreateOrder = this.payPalCreateOrder.bind(this)
+    this.payPalOnApprove = this.payPalOnApprove.bind(this)
+    this.handleSetPaymentMethod = this.handleSetPaymentMethod.bind(this)
+  }
+
+  /**
+   * Returns route for next checkout section  
+   */
+  nextSection () {
+    Router.push('/checkout/shipping-address')
+  }
+
+  /**
+   * Sets the current checkout page title  
+   */
+  pageTitle = () => 'Payment Method'
+
+
+  /**
+   * Sets the current checkout page step 
+   */
+  currentStep = () => 1
+
+  /**
+   * Returns the props for the continueButton
+   */
+  continueButtonProps () {
+    return null
+  }
+
+  /**
+   * When a payment method is selected, this funtion is called and 
+   * it sets the current payment method in the state
+   * @param  {object} paymentMethod
+   */
+  handleSetPaymentMethod (paymentMethod) {
+    this.props.dispatch(setPaymentMethod(paymentMethod))
+  }
+
+  /**
+   * When the buyer clicks the PayPal button, this function is called and
+   * it handles the creation of the order 
+   * @param  {object} data
+   * @param  {object} actions
+   */
+  payPalCreateOrder(data, actions) {
+    const { cart } = this.props
+    return actions.order.create({
+      purchase_units: [{
+        // currency is specified in the PayPal SDK script
+        amount: {
+          value: cart.total
+        }
+      }]
+    })
+  }
+
+  /**
+   * After the cusrtomer approves the transaction on paypal.com, 
+   * this function is called and it retrieves the transaction details
+   * @param  {object} data
+   * @param  {object} actions
+   */
+  payPalOnApprove(data, actions) {
+    return actions.order.get().then((order) => this.handlePayPalOrderResponse(order))
+  }
+
+  /**
+   * Handles data for the fetched PayPal Order 
+   * @param  {object} order
+   */
+  handlePayPalOrderResponse (order) {
+    const payer = order.payer
+    const payerEmail = payer.email_address
+    const payerPhoneNumber = payer.phone.phone_number.national_number
+    const shippingDetails = order.purchase_units[0].shipping
+    const shippingFullName = shippingDetails.name.full_name.split(' ')
+    // set loading to true as we handle the order information
+    this.setState({ loading: true })
+    // handle basic order details - id, intent, status + create_time
+    this.handlePayPalOrderDetails(order)
+    // handle parsing + setting of billing address in state + creation
+    this.handleBillingAddressCreation(
+      this.parsePayPalAddress(
+        payer.name.given_name,
+        payer.name.surname,
+        payerEmail,
+        payerPhoneNumber,
+        payer.address
+      )
+    )
+    // handle parsing + setting of shipping address in state + creation
+    this.handleShippingAddressCreation(
+      this.parsePayPalAddress(
+        shippingFullName.slice(0, -1).join(' '),
+        shippingFullName.slice(-1).join(' '),
+        payerEmail,
+        payerPhoneNumber,
+        shippingDetails.address
+      )
+    ).then(() => {
```

Comment:
Should `.then` happen after _both_ `handleBillingAddressCreation` and `handleShippingAddressCreation` complete? Is it worth wrapping them in `Promise.all`?

Reply:
Yes, I've addressed this in then next PR -  Part 2 👍